eng/tools/versioning/increment.js

```diff
@@ -0,0 +1,54 @@
+let argv = require("yargs")
+  .options({
+    "artifact-name": {
+      type: "string",
+      describe:
+        "name of the artifact to be incremented (e.g. azure-keyvault-secrets), will be translated to @azure/(package) format",
+      demandOption: true
+    },
+    "repo-root": {
+      type: "string",
+      describe: "root of the repository (e.g. ../../../)",
+      demandOption: true
+    }
+  })
+  .help().argv;
+
+const path = require("path");
+const semver = require("semver");
+const versionUtils = require("./VersionUtils");
+
+function incrementVersion(currentVersion) {
+  const prerelease = semver.prerelease(currentVersion);
+  if (prerelease) {
+    return semver.inc(currentVersion, "prerelease");
+  }
+
+  return `${semver.inc(currentVersion, "minor")}-preview.1`;
+}
+
+async function main(argv) {
+  const artifactName = argv["artifact-name"];
+  const repoRoot = argv["repo-root"];
+
+  const packageName = artifactName.replace("azure-", "@azure/");
+  const rushSpec = await versionUtils.getRushSpec(repoRoot);
+
+  const targetPackage = rushSpec.projects.find(
+    packageSpec => packageSpec.packageName == packageName
+  );
+  const targetFile = path.resolve(
+    path.join(repoRoot, targetPackage.projectFolder, "package.json")
+  );
+
+  const targetFileContents = await versionUtils.readFileJson(targetFile);
```

Comment:
This seems to only update the project file do we intend to update things like readmes as well?

Reply:
From conversations today: 
* Version numbers shouldn't be (and for the most part aren't) in README.md files. So we shouldn't build automation for that. 
* Version numbers do show up in constant files (e.g. https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/src/utils/constants.ts#L4) so we should make sure they're properly updated (cursory research shows some are out of sync)... following up with @daviwil to get more info
* It makes sense to add to the changelog files (#5651)

Reply:
In an ideal world, I think we'd eliminate the version constants altogether or at the very least put them in a standard location in every package. We shouldn't have to specify them again in the code when they are already specified in package.json.

I was discussing with @chradek earlier today about possibly using a rollup plugin that would read the package.json at build-time, drop all the unnecessary keys from the package object, and use that instead of a static module file with the version exported as a string literal.

Reply:
If the version shows up in one place then it makes all this code rather trivial... I am expanding what this does on another branch (because it's easier to test unified pipelines when pushing to upstream instead of a fork) 

https://github.com/Azure/azure-sdk-for-js/pull/5652

We've accumulated a few entertaining edge cases that would be pretty well eliminated if we could get versions in one location: 
* generated code that doesn't get generated after every release (makes it hard to tag things as they exist today because the code generator will blow away the tags until it gets fixed) 
* constants peppered in files with varying conventions (`version`, `packageVersion`, `coreArmVersion`, etc.) 
* constants whose values "look like" semver strings (e.g. ip addresses)